config/reader.go

```diff
@@ -49,29 +57,150 @@ func getAgentInfo() (*apppb.AgentInfo, error) {
 	}, nil
 }
 
-var viamDotDir = filepath.Join(os.Getenv("HOME"), ".viam")
+// FileReader responsible for converting the raw file into a Config type
+type FileReader struct {
+	// filepath is the filepath of the config
+	Filepath string
+	// config is the output of the reader and what gets manipulated
+	Config *Config
 
-func getCloudCacheFilePath(id string) string {
-	return filepath.Join(viamDotDir, fmt.Sprintf("cached_cloud_config_%s.json", id))
+	fileInBytes      []byte
+	localDataDir     string
+	localPackagesDir string // path.Join(viamDotDir, "packages")
+
+	// CurrentPackages stores a list of valid packages used to try and update placeholders in the
+	// entire config
+	CurrentPackages map[string]string
+}
+
+// copied from package-manager
+func HashName(f PackageConfig) string {
+	// replace / to avoid a directory path in the name. This will happen with `org/package` format.
+	// also makes the path valid by replacing all the dots with _
+	return fmt.Sprintf("%s-%s", strings.ReplaceAll(f.Package, "/", "-"), HashVersion(f.Version))
+}
```

Comment:
Suggest moving to a common utils location so that package manager and reader never get out of sync. We might unintentionally change in only one place in the future.

Reply:
moved this to the config file, we have a lot of utility functions there already so thought this made sense.